app/(candidate)/dashboard/door-knocking/components/InteractionsByDay.js

```diff
@@ -0,0 +1,108 @@
+'use client';
+import H2 from '@shared/typography/H2';
+import Caption from '@shared/typography/Caption';
+import Paper from '@shared/utils/Paper';
+import { numberFormatter } from 'helpers/numberHelper';
+import { Bar } from 'react-chartjs-2';
+import interactionsColors from './interactionsColors';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+} from 'chart.js';
+import { slugify } from 'helpers/articleHelper';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+  Legend,
+);
+
+export default function InteractionsSummaryPie({ summary }) {
+  const { totalInteractions, interactionsByDay } = summary || {};
+
+  // Process data for the chart
+  const processChartData = () => {
+    if (!interactionsByDay) return null;
+
+    // Get all unique interaction types
+    const allTypes = new Set();
+    Object.values(interactionsByDay).forEach((dayData) => {
+      Object.keys(dayData).forEach((key) => {
+        if (key !== 'count') allTypes.add(key);
+      });
+    });
+    const interactionTypes = Array.from(allTypes);
+
+    // Generate array of last 30 days
+    const dates = [];
+    const today = new Date();
+    for (let i = 29; i >= 0; i--) {
+      const date = new Date(today);
+      date.setDate(today.getDate() - i);
+      dates.push(date.toISOString().split('T')[0]);
+    }
+
+    // Format dates for display
+    const labels = dates.map((date) => new Date(date).toLocaleDateString());
+
+    // Create datasets with zero-filled data for missing days
+    const datasets = interactionTypes.map((type) => ({
+      label: type,
+      data: dates.map((date) => {
+        const dayData = interactionsByDay[date];
+        return dayData ? dayData[type] || 0 : 0;
+      }),
+      backgroundColor: interactionsColors[slugify(type, true)] || '#CBD5E1', // Fallback color if not found
+    }));
+
+    return {
+      labels,
+      datasets,
+    };
+  };
```

Comment:
Should be able to move this out into a util method, so as not to clutter up the component w/ all this business logic?

Looks like the only dependency here is interactionsByDay and that could just be passed in via method argument instead of using closure, no?